packages/flutter/lib/src/rendering/error.dart

```diff
@@ -82,39 +77,87 @@ class RenderErrorBox extends RenderBox {
     size = constraints.constrain(const Size(_kMaxWidth, _kMaxHeight));
   }
 
+  /// The distance to place around the text.
+  ///
+  /// This is intended to ensure that if the [RenderErrorBox] is placed at the top left
+  /// of the screen, under the system's status bar, the error text is still visible in
+  /// the area below the status bar.
+  ///
+  /// The padding is ignored if the error box is smaller than the padding.
+  ///
+  /// See also:
+  ///
+  ///  * [minimumWidth], which controls how wide the box must be before the
+  //     horizontal padding is applied.
+  static EdgeInsets padding = const EdgeInsets.fromLTRB(64.0, 96.0, 64.0, 12.0);
+
+  /// The width below which the horizontal padding is not applied.
+  ///
+  /// If the left and right padding would reduce the available width to less than
+  /// this value, then the text is rendered flush with the left edge.
+  static double minimumWidth = 200.0;
+
   /// The color to use when painting the background of [RenderErrorBox] objects.
-  static Color backgroundColor = const Color(0xF0900000);
+  ///
+  /// Defaults to red in debug mode, a light gray otherwise.
+  static Color backgroundColor = _initBackgroundColor();
+
+  static Color _initBackgroundColor() {
+    Color result = const Color(0xF0C0C0C0);
+    assert(() {
```

Comment:
To make the intend of these asserts clearer: Maybe use kReleaseMode or kProfileMode instead?

Reply:
I'd really rather we got rid of those constants entirely...

Reply:
To me, the constants make the intent clearer and the code more readable then the assert wrapping.

Reply:
if statements look dynamic. I want something that looks like an `#ifdef` (but structural).